src/platform.cpp

```diff
@@ -668,6 +683,48 @@ internal bool DebugWriteEntireFile(char const *Filename, char const *Content, u6
 }
 #endif
 
+#if HITMAN_INTERNAL
+internal void DebugRecordInput(debug_input_recording *InputRecorder, game_input *NewInput, game_memory *GameMemory)
+{
+    // Setup Input recording, 
+    if (InputRecorder->RecordHandle == NULL)
+    {
+        InputRecorder->RecordHandle = fopen("../data/input.hmi", "w");
+        u64 TotalMemorySize = InputRecorder->TotalMemorySize;
+
+        u32 BytesToWrite = (u32)InputRecorder->TotalMemorySize;
+        Assert(TotalMemorySize == BytesToWrite); // This can't be more then 4Gb, because then we cannot write it to a file at once.
+        u64 Written = fwrite(GameMemory->PermanentStorage, 1, TotalMemorySize, InputRecorder->RecordHandle);
+        Assert(BytesToWrite == Written);
+    }
+    u64 InputSize = sizeof(*NewInput);
+    u64 Written = fwrite(NewInput, 1, InputSize, InputRecorder->RecordHandle);
+    Assert(InputSize == Written);
+}
+
+internal void DebugPlaybackInput(debug_input_recording *InputRecorder, game_input *NewInput, game_memory *GameMemory)
+{
+    if (InputRecorder->RecordHandle != NULL)
+    {
+        fclose(InputRecorder->RecordHandle);
+        InputRecorder->RecordHandle = NULL;
+    }
+    if (InputRecorder->PlaybackHandle == NULL)
+    {
+        InputRecorder->PlaybackHandle = fopen("../data/input.hmi", "r");
+        u32 BytesToRead = (u32)InputRecorder->TotalMemorySize;
+        Assert(InputRecorder->TotalMemorySize == BytesToRead); // This can't be more then 4Gb on Windows with Live loop, because we cannot write it to a file at once.
+        fread(GameMemory->PermanentStorage, BytesToRead, 1, InputRecorder->PlaybackHandle);
+    }
+    u64 InputSize = sizeof(game_input);
+    u64 Read = fread(NewInput, 1, InputSize, InputRecorder->PlaybackHandle);
```

Comment:
What happens if we break playback of in the middle of a loop? File is not closed and on next playback might use some old input recording still in the file? Investigate and fix!